spec/integration/parser/compiler_spec.rb

```diff
@@ -522,4 +522,31 @@ class xport_test {
       expect(catalog).to have_resource("Notify[nbr2]").with_parameter(:message, 'overridden')
     end
   end
+
+
+  context "when converting catalog to resource" do
+    it "the same environment is used for compilation as for transformation to resource form" do
+        Puppet[:code] = <<-MANIFEST
+          notify { 'dummy':
+          }
+        MANIFEST
+
+        # Create an instance of a mocha state machine that checks that the environment is
+        # 'production'
+        in_production_state = Class.new do
+
+          def active?
+            Puppet.lookup(:current_environment).name == :production
+          end
+
+          def to_s
+            # displayed if the expectation about this state fails
+            "in production environment"
+          end
+        end.new
```

Comment:
@hlindberg I think it's fine as is. I thought it started in production environment. As for the mocha bits, it's a little concerning that knows about mocha internals like that. Let me take a look and see if I can come up with something...